Invalid JSON config throws panic

While working on a project I had this invalid JSON configuration file:

```json
{
  "consul": "{{ .CONSUL }}:8500",
  "services": [
    {
      "name": "app",
      "port": 8000,
      "health": ["/usr/local/bin/manage.sh", "health"],
      "poll": 5,
      "ttl": 15,
    }
  ]
}
```

The error is the trailing comma on the `ttl` field... JSON is awesome like that. But instead of returning an error and exiting cleanly, we panic and give the end user a very unfriendly stack trace. Here's the running goroutine:

```
panic: runtime error: makeslice: len out of range

goroutine 1 [running]:
strings.Repeat(0xb3bac8, 0x1, 0xfffffffffffffffe, 0x0, 0x0)
        /usr/local/go/src/strings/strings.go:464 +0x5f
containerbuddy.highlightError(0xc820126000, 0x171, 0x1dc, 0xc5, 0xc820128020, 0x7d, 0x0, 0x0)
        /go/cb/src/containerbuddy/config.go:400 +0x4fc
containerbuddy.newJSONParseError(0xc820126000, 0x171, 0x1dc, 0xc8200dec40, 0x0, 0x0)
        /go/cb/src/containerbuddy/config.go:380 +0x6d
containerbuddy.unmarshalConfig(0xc820126000, 0x171, 0x1dc, 0xc820126000, 0x0, 0x0)
        /go/cb/src/containerbuddy/config.go:374 +0x241
containerbuddy.parseConfig(0xc82001218f, 0x1f, 0xc82001218f, 0x0, 0x0)
        /go/cb/src/containerbuddy/config.go:362 +0x542
containerbuddy.loadConfig(0x1, 0x0, 0x0)
        /go/cb/src/containerbuddy/config.go:199 +0x303
containerbuddy.Main()
        /go/cb/src/containerbuddy/main.go:20 +0x3d
main.main()
        /go/cb/main.go:6 +0x14

```

--- a/config/config.go
+++ b/config/config.go
@@ -277,11 +277,17 @@ func highlightError(data []byte, pos int64) (int, int, string) {
 		readBytes := int64(len(scanner.Bytes()))
 		offset += readBytes
 		if offset >= pos-1 {
-			highlight = fmt.Sprintf("%s^", strings.Repeat("-", int(7+col-1)))
+			count := int(7 + col - 1)
+			if count > 0 {
+				highlight = fmt.Sprintf("%s^", strings.Repeat("-", count))
+			}
 			break
 		}
 		col -= readBytes + 1
 		line++
+	}
+	if col < 0 {
+		highlight = "Do you have an extra comma somewhere?"
 	}
 	return line, int(col), fmt.Sprintf("%s%s%s", prevLine, thisLine, highlight)
 }

--- a/config/config_test.go
+++ b/config/config_test.go
@@ -176,6 +176,18 @@ func TestJSONTemplateParseError2(t *testing.T) {
 		"Parse error at line:col [5:5]")
 }
 
+func TestParseTrailingComma(t *testing.T) {
+	defer argTestCleanup(argTestSetup())
+	testParseExpectError(t,
+		`{
+			"consul": "consul:8500",
+			"tasks": [{
+				"command": ["echo","hi"]
+			},
+		]
+	}`, "Do you have an extra comma somewhere?")
+}
+
 func TestMetricServiceCreation(t *testing.T) {
 
 	jsonFragment := []byte(`{

